rust_snuba/src/processors/outcomes.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashSet;
+
+use crate::types::{BadMessage, BytesInsertBatch, KafkaMessageMetadata};
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use serde::Deserialize;
+
+pub fn process_message(
+    payload: KafkaPayload,
+    _metadata: KafkaMessageMetadata,
+) -> Result<BytesInsertBatch, BadMessage> {
+    let payload_bytes = payload.payload.ok_or(BadMessage)?;
+
+    Ok(BytesInsertBatch { rows: vec![] })
+}
+
+#[derive(Debug, Default, Deserialize)]
+struct FromOutcomeMessage {
```

Comment:
Did you consider merging `FromOutcomeMessage` and `Outcome`. The keys are essentially identical, so I wonder if you really need 2 structs at all